lib/logstash/filters/mutate.rb

```diff
@@ -338,85 +334,91 @@ def gsub_dynamic_fields(event, original, needle, replacement)
     end
   end
 
-  private
   def uppercase(event)
     @uppercase.each do |field|
-      if event[field].is_a?(Array)
-        event[field].each { |v| v.upcase! }
-      elsif event[field].is_a?(String)
-        event[field].upcase!
-      else
-        @logger.debug("Can't uppercase something that isn't a string",
-                      :field => field, :value => event[field])
-      end
+      original = event[field]
+      event[field] = case original
+        when Array
+          original.map(&:upcase!)
+        when String
+          original.upcase!
+        else
+          @logger.debug("Can't uppercase something that isn't a string",
+                        :field => field, :value => original)
+          original
+        end
     end
   end # def uppercase
 
-  private
   def lowercase(event)
     @lowercase.each do |field|
-      if event[field].is_a?(Array)
-        event[field].each { |v| v.downcase! }
-      elsif event[field].is_a?(String)
-        event[field].downcase!
-      else
-        @logger.debug("Can't lowercase something that isn't a string",
-                      :field => field, :value => event[field])
-      end
+      original = event[field]
+      event[field] = case original
+        when Array
+          original.map(&:downcase!)
+        when String
+          original.downcase!
+        else
+          @logger.debug("Can't lowercase something that isn't a string",
+                        :field => field, :value => original)
+          original
+        end
     end
   end # def lowercase
 
-  private
   def split(event)
     @split.each do |field, separator|
-      if event[field].is_a?(String)
-        event[field] = event[field].split(separator)
+      original = event[field]
+      if original.is_a?(String)
+        event[field] = original.split(separator)
       else
         @logger.debug("Can't split something that isn't a string",
-                      :field => field, :value => event[field])
+                      :field => field, :value => original)
       end
     end
   end
 
-  private
   def join(event)
     @join.each do |field, separator|
-      if event[field].is_a?(Array)
-        event[field] = event[field].join(separator)
+      original = event[field]
```

Comment:
@jsvd True.  However I had the change of [] to get(field) and []= to set(field, value) in mind when I changed all to use the original local variable as it reads better and is consistent across classes.
e.g.
```ruby
original = event.get[field]
mutated = original.map(&:to_i)
event.set(field, mutated)
```
as opposed to
```ruby
event.set(field, event.get(field).map(&:to_i))
```